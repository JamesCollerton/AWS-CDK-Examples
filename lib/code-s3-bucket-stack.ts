import * as cdk from '@aws-cdk/core';
import * as s3 from '@aws-cdk/aws-s3';
import { RemovalPolicy } from '@aws-cdk/core';

export class CodeS3BucketStack extends cdk.Stack {
  constructor(scope: cdk.Construct, id: string, props?: cdk.StackProps) {
    super(scope, id, props);

    // The code that defines your stack goes here
    new s3.Bucket(this, 'CodeBucket', {
      bucketName: "react-ecs-bucket",
      removalPolicy: RemovalPolicy.DESTROY,
      versioned: true
    });
  }
}
