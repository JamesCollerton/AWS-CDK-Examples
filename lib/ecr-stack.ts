import * as cdk from '@aws-cdk/core';
import * as ecr from '@aws-cdk/aws-ecr';
import { RemovalPolicy } from '@aws-cdk/core';


export class EcrStack extends cdk.Stack {
  constructor(scope: cdk.Construct, id: string, props?: cdk.StackProps) {
    super(scope, id, props);

    new ecr.Repository(this, 'EcrRepository', {
        repositoryName: "react-ecs",
        removalPolicy: RemovalPolicy.DESTROY
    });
  }
}
