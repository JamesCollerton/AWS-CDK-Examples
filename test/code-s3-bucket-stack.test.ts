import { expect as expectCDK, SynthUtils } from '@aws-cdk/assert';
import * as cdk from '@aws-cdk/core';
import * as AwsCdkExamples from '../lib/code-s3-bucket-stack';

test('S3 Bucket Snapshot', () => {
    const app = new cdk.App();
    const stack = new AwsCdkExamples.CodeS3BucketStack(app, 'S3BucketSnapshotTest');
    expect(SynthUtils.toCloudFormation(stack)).toMatchSnapshot();
});
