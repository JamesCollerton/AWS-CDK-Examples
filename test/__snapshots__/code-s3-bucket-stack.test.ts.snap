// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`S3 Bucket Snapshot 1`] = `
Object {
  "Resources": Object {
    "CodeBucketFF4C7AD6": Object {
      "DeletionPolicy": "Delete",
      "Properties": Object {
        "BucketName": "react-ecs-bucket",
        "VersioningConfiguration": Object {
          "Status": "Enabled",
        },
      },
      "Type": "AWS::S3::Bucket",
      "UpdateReplacePolicy": "Delete",
    },
  },
}
`;
